### PR TITLE
Pe 12553

### PR DESCRIPTION
**Pull request by shriganeshs**
This is a pre-PR for
https://jira.inside-box.net/browse/PE-12552
https://jira.inside-box.net/browse/PE-12553

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @shriganeshs has signed the CLA. Thanks for the pull request!

File: app/master/build.py [hunk 1]
```diff
@@ -309,6 +289,35 @@ def _on_enter_error_state(self, event):
...
+        self._logger.info('Successfully fetched project for build {}.', self._build_id)
+
+        job_config = self.project_type.job_config()
+        if job_config is None:
+            raise RuntimeError('Build failed while trying to parse clusterrunner.yaml.')
+
+        subjob_calculator = getattr(event, 'subjob_calculator', None)
+        if subjob_calculator is None:
+            raise RuntimeError('Build failed as subjob_calculator is not set for this build {}.', self._build_id)
+        subjobs = subjob_calculator.compute_subjobs_for_build(self._build_id, job_config, self.project_type)
```
Comment (shriganeshs) [inline app/master/build.py]: When the build is canceled during atomizing, an exception `AtomizerError` is raised which causes stack trace in logs. I am still working on avoiding the stack trace when the build is canceled.
I tried catching and ignoring `AtomizerError` if the build is canceled (see below code), but that leaves the build in `PREPARING` state.

```python
        try:
            subjobs = subjob_calculator.compute_subjobs_for_build(self._build_id, job_config, self.project_type)
        except AtomizerError:
            if self.is_canceled:
                return
            else:
                raise
```

File: app/master/build.py [hunk 2]
```diff
@@ -147,31 +149,9 @@ def prepare(self, subjob_calculator):
...
-
-        for subjob in subjobs:
-            self._all_subjobs_by_id[subjob.subjob_id()] = subjob
-            self._unstarted_subjobs.put(subjob)
-
-        self._timing_file_path = self._project_type.timing_file_path(job_config.name)
-        app.util.fs.create_dir(self._build_results_dir())
-        self._state_machine.trigger(BuildEvent.FINISH_PREPARE)
+        # SS_DEBUG
+        time.sleep(10)
```
Comment (shriganeshs) [inline app/master/build.py]: This sleep is just for testing. I will remove it later.

Comment (wjdhollow) [inline app/master/build.py]: Oh, is SS for your initials?

**Pull request closed without merging**